### DISCUSSION TITLE
Is it possible to make multiple medication administrations for a given medication?

### DISCUSSION THREAD

**Discussion by mduna** [synthetichealth/synthea#1077] [Q&A]
Is it possible to make multiple medication administrations for a given medication? For example, a vancomycin IV medication once a day for 5 days. Synthea will generate five MedicationRequest resources for five days, one resource per day. If it is possible, could please show me an example? Thanks.

**Comment by jawalonoski** [comment]
It should be possible to do this. Below is an example of a Module that *should* work.

However, upon testing it, it does not seem to be functioning correctly. Which means we have a bug somewhere.

![image](https://user-images.githubusercontent.com/1054765/171186098-202ba2a3-f0db-4e7d-a6b0-23b97572bdc6.png)

**Reply by jawalonoski** [reply]
I made an issue #1079

**Reply by mduna** [reply]
Hi Jason, Thank your quick response. Follow your example, it does not work for me. Please see my json file. Not sure what I did wrong. Your help will be appreciated. 

{
  "name": "cot-cdi2",
  "remarks": [
    "A blank module"
  ],
  "states": {
    "Initial": {
      "type": "Initial",
      "direct_transition": "Jan2022"
    },
    "Jan2022": {
      "type": "Guard",
      "allow": {
        "condition_type": "And",
        "conditions": [
          {
            "condition_type": "Date",
            "operator": ">",
            "date": {
              "year": 2022,
              "month": 1,
              "day": 1,
              "hour": 0,
              "minute": 0,
              "second": 0,
              "millisecond": 0
            }
          },
          {
            "condition_type": "Date",
            "operator": "<=",
            "date": {
              "year": 2022,
              "month": 2,
              "day": 1,
              "hour": 0,
              "minute": 0,
              "second": 0,
              "millisecond": 0
            },
            "value": 0
          }
        ]
      },
      "direct_transition": "Wait_1_Month"
    },
    "Encounter_End": {
      "type": "EncounterEnd",
      "direct_transition": "Wait_1_Month"
    },
    "In_hospital_delay": {
      "type": "Delay",
      "direct_transition": "lab-stool",
      "exact": {
        "quantity": 2,
        "unit": "days"
      }
    },
    "Hospital inpatient Admission_1": {
      "type": "Encounter",
      "encounter_class": "inpatient",
      "reason": "",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": 1505002,
          "display": "SNOMED Code"
        }
      ],
      "direct_transition": "In_hospital_delay"
    },
    "Examplitis": {
      "type": "ConditionOnset",
      "target_encounter": "Hospital inpatient Admission_1",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": 195967001,
          "display": "Asthma"
        }
      ],
      "direct_transition": "Hospital inpatient Admission_1"
    },
    "lab-stool": {
      "type": "Observation",
      "category": "laboratory",
      "unit": "mg/dL",
      "codes": [
        {
          "system": "LOINC",
          "code": "10895-1",
          "display": "Clostridioides difficile toxin B [Presence] in Stool"
        }
      ],
      "direct_transition": "Counter",
      "value_code": {
        "system": "SNOMED-CT",
        "code": 719220002,
        "display": "Positive"
      }
    },
    "vancomycin1": {
      "type": "MedicationOrder",
      "codes": [
        {
          "system": "RxNorm",
          "code": 239209,
          "display": "vancomycin 100 MG/ML Injectable Solution"
        }
      ],
      "direct_transition": "Check_Counter",
      "prescription": {
        "dosage": {
          "amount": 1,
          "frequency": 1,
          "period": 1,
          "unit": "days"
        },
        "duration": {
          "quantity": 1,
          "unit": "days"
        },
        "refills": 5
      },
      "administration": true
    },
    "Counter": {
      "type": "SetAttribute",
      "attribute": "counter",
      "direct_transition": "vancomycin1",
      "value": 1
    },
    "Check_Counter": {
      "type": "Simple",
      "conditional_transition": [
        {
          "transition": "Increment_Counter",
          "condition": {
            "condition_type": "Attribute",
            "attribute": "counter",
            "operator": "<",
            "value": 5
          }
        },
        {
          "transition": "Encounter_End"
        }
      ]
    },
    "Increment_Counter": {
      "type": "Counter",
      "attribute": "counter",
      "action": "increment",
      "direct_transition": "Wait_1_Day"
    },
    "Wait_1_Day": {
      "type": "Delay",
      "distribution": {
        "kind": "EXACT",
        "parameters": {
          "value": 1
        }
      },
      "unit": "days",
      "direct_transition": "vancomycin1"
    },
    "Wait_1_Month": {
      "type": "Delay",
      "distribution": {
        "kind": "EXACT",
        "parameters": {
          "value": 1
        }
      },
      "unit": "months",
      "direct_transition": "Examplitis"
    }
  },
  "gmf_version": 2

**Reply by jawalonoski** [reply]
I did not look at your module.

There is a bug, which is why I opened a new issue #1079

**Comment by mduna** [comment]
Hi Jason, Ok, I will wait for the bug fixes, then give a try. Thanks.
